aten/src/THC/THCTensorRandom.cu

```diff
@@ -6,12 +6,35 @@
 #include "THCReduceApplyUtils.cuh"
 #include "THCTensorRandom.cuh"
 #include "THCGenerator.hpp"
+#include "ATen/Config.h"
 
 #include <thrust/functional.h>
 #include <curand.h>
 #include <curand_kernel.h>
+
+// Forward declarations of functions that are defined in libcurand_static.a
+// This is to avoid multiple-definitions of these when statically linking
+// cudarand in both Caffe2 and ATen
+#if CAFFE2_STATIC_LINK_CUDA()
+curandStatus_t curandMakeMTGP32Constants(const mtgp32_params_fast_t params[],mtgp32_kernel_params_t * p);
+void mtgp32_init_state(unsigned int state[],
+              const mtgp32_params_fast_t *para, unsigned int seed);
```

Comment:
weird indent?

Reply:
Oh right. I was going to change that but forgot